test/metabase/test/data/impl.clj

```diff
@@ -110,7 +110,7 @@
         (u/with-timeout sync-timeout-ms
           (let [reference-duration (or (some-> (get @reference-sync-durations database-name) u/format-nanoseconds)
                                        "NONE")
-                quick-sync? (not= database-name "test-data")]
+                quick-sync? ((complement #{"test-data" "sample-dataset"}) database-name)]
```

Comment:
This should be the only strange thing in this PR. `metabase.query-processor.util.add-alias-info-test/join-source-query-join-test` tests a query shape and one of the fields has `:semantic_type :type/Category`. But the infra used to only run a full sync on "test-data". Which means some fields wouldn't have semantic_types unless some other test randomly did a full sync.